orchagent/srv6orch.cpp

```diff
@@ -600,6 +784,322 @@ bool Srv6Orch::deleteMysidEntry(const string my_sid_string)
+    srv6_p2p_tunnel_table_[k].tunnel_map_id = srv6_tunnel_map_id;
+    return true;
+}
+
+bool Srv6Orch::deleteSrv6P2pTunnel(const P2pTunnelKey &k)
+{
+    if (srv6_p2p_tunnel_table_.find(k) == srv6_p2p_tunnel_table_.end())
+    {
+        return true;
+    }
+
+    sai_status_t status;
+
+    // 0. remove tunnel
+    status = sai_tunnel_api->remove_tunnel(srv6_p2p_tunnel_table_[k].tunnel_id);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to remove SRV6 p2p tunnel object for src_ip %s, dst_ip: %s", k.src_ip.c_str(), k.endpoint.c_str());
+        return false;
+    }
+
+    // 1. remove tunnel map
+    status = sai_tunnel_api->remove_tunnel_map(srv6_p2p_tunnel_table_[k].tunnel_map_id);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to remove SRV6 tunnel map object for src_ip %s, dst_ip: %s", k.src_ip.c_str(), k.endpoint.c_str());
+        return false;
+    }
+
+    srv6_p2p_tunnel_table_.erase(k);
+    return true;
+}
+
+void Srv6Orch::srv6P2ptunnelUpdateNexthops(const P2pTunnelKey &k, const NextHopKey &nhkey, bool insert)
+{
+    if (insert)
+    {
+        srv6_p2p_tunnel_table_[k].nexthops.insert(nhkey);
+    }
+    else
+    {
+        srv6_p2p_tunnel_table_[k].nexthops.erase(nhkey);
+    }
+}
+
+size_t Srv6Orch::srv6P2pTunnelNexthopSize(const P2pTunnelKey &k)
+{
+    return srv6_p2p_tunnel_table_[k].nexthops.size();
+}
+
+bool Srv6Orch::createSrv6Vpn(const NextHopKey &nh, const uint32_t prefix_agg_id)
+{
+    SWSS_LOG_ENTER();
+
+    sai_status_t status;
+
+    Srv6TunnelMapEntryKey tmek;
+    tmek.src_ip = nh.srv6_source;
+    tmek.endpoint = nh.ip_address.to_string();
+    tmek.vpn_sid = nh.srv6_vpn_sid;
+    tmek.prefix_agg_id = prefix_agg_id;
+
+    if (srv6_tunnel_map_entry_table_.find(tmek) != srv6_tunnel_map_entry_table_.end())
+    {
+        srv6_tunnel_map_entry_table_[tmek].ref_count++;
+        return true;
+    }
+
+    P2pTunnelKey k;
+    k.src_ip = nh.srv6_source;
+    k.endpoint = nh.ip_address.to_string();
+
+    // Accessibility is guaranteed by external
+    sai_object_id_t tunnel_map_id = srv6_p2p_tunnel_table_[k].tunnel_map_id;
+
+    // 1. create vpn tunnel_map entry
+    sai_attribute_t tunnel_map_entry_attr;
+    vector<sai_attribute_t> tunnel_map_entry_attrs;
+    sai_object_id_t tunnel_entry_id;
+
+    tunnel_map_entry_attr.id = SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE;
+    tunnel_map_entry_attr.value.u32 = SAI_TUNNEL_MAP_TYPE_PREFIX_AGG_ID_TO_SRV6_VPN_SID;
+    tunnel_map_entry_attrs.push_back(tunnel_map_entry_attr);
+
+    tunnel_map_entry_attr.id = SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP;
+    tunnel_map_entry_attr.value.oid = tunnel_map_id;
+    tunnel_map_entry_attrs.push_back(tunnel_map_entry_attr);
+
+    tunnel_map_entry_attr.id = SAI_TUNNEL_MAP_ENTRY_ATTR_PREFIX_AGG_ID_KEY;
+    tunnel_map_entry_attr.value.u32 = prefix_agg_id;
+    tunnel_map_entry_attrs.push_back(tunnel_map_entry_attr);
+
+    IpAddress vpn_sid(nh.srv6_vpn_sid);
```

Comment:
Do we have any sonic-utility CLI to dump these map entries? We have VLAN-->VNI, VRF-->VNI map entries under 'show vxlan'. We could have one more for SRV6 tunnel map entries.

Reply:
sure, I can add the CLI in sonic-utility after this PR merged.